GitUI/BranchTreePanel/RepoObjectsTree.Nodes.cs

```diff
@@ -152,8 +157,15 @@ private void UICommands_PostRepositoryChanged(object sender, GitUIPluginInterfac
 
             protected abstract void PostRepositoryChanged();
 
+            public void StartAsyncRefreshTree()
+            {
+                ThreadHelper.AssertOnUIThread();
```

Comment:
❓ Why?

* CancellationTokenSequence is safe for concurrent use
* RefreshTreeAsync is asynchronous

A UI thread dependency here is a strong indicator of a bug.

Reply:
I had issues with accessing CancellationTokenSequence after disposing it, so I reworked the code so that it was always accessed on the main thread. At first, I wanted to see if there was a way to query that CTS was disposed of, but there wasn't. I considered try-catching the DisposedException, but that just seemed wrong. In his PR, @jbialobr made changes to CancellationTokenSequence so that calling Next on it throws OperationCancelException, which is a better solution.